redis_schedule_job.go

```diff
@@ -107,24 +108,87 @@ func (t *scheduleJob) enqueue(ctx context.Context, msg *Message) error {
 	return err
 }
 
-func (t *scheduleJob) doConsume(ctx context.Context, max string, channel, topic string) error {
-	zRangeBy := &redis.ZRangeBy{
-		Min: defaultScheduleJobConfig.scoreMin,
-		Max: max,
-	}
-	key := MakeZSetKey(t.broker.prefix, channel, topic)
+var (
+	Unexpired = errors.New("unexpired")
+)
 
-	val := t.broker.client.ZRevRangeByScore(ctx, key, zRangeBy).Val()
+func (t *scheduleJob) run(ctx context.Context, channel, topic string, closeCh chan struct{}) error {
 
-	if len(val) <= 0 {
-		return nil
+	var (
+		timeUnit      = MakeTimeUnit(t.broker.prefix, channel, topic)
+		currentString string // millisecond string
+	)
+
+	timer := timex.TimerPool.Get(1 * time.Second)
+	defer timex.TimerPool.Put(timer)
+
+	for {
+		select {
+		case <-closeCh:
+			t.broker.pool.Release()
```

Comment:
It is best not to release the broker's pool here and leave it to the broker for processing, because we don't know whether it is still used elsewhere. that's my understanding.

Reply:
repeat to release ant.pool.
have fixed.